Source/Fuse.Marshal/Marshal.Cast.uno

```diff
@@ -173,34 +208,43 @@ namespace Fuse
 				var s = (string)o;
 				if (s.StartsWith("#"))
 				{
-					value = Uno.Color.FromHex(s);
-					size = 4;
-					return true;
+					//TODO: once https://github.com/fusetools/uno/pull/1383 is avialble use Color.TryParse instead
+					try
+					{
+						value = Uno.Color.FromHex(s);
+						size = 4;
+						return true;
+					}
+					catch (Exception ex)
+					{
+						return false;
+					}
 				}
 			}
 			else if (o is IArray)
 			{
 				var a = (IArray)o;
-				var x = a.Length > 0 ? Marshal.ToFloat(a[0]) : 0.0f;
-				var y = a.Length > 1 ? Marshal.ToFloat(a[1]) : 0.0f;
-				var z = a.Length > 2 ? Marshal.ToFloat(a[2]) : 0.0f;
-				var w = a.Length > 3 ? Marshal.ToFloat(a[3]) : 0.0f;
+				float x = 0,y = 0,z = 0,w =0;
+				if (!TryToFloat( a[0], out x ) ||
+					!TryToFloat( a[1], out y ) ||
+					!TryToFloat( a[2], out z ) ||
+					!TryToFloat( a[3], out w ))
```

Comment:
This looks like it will cause IndexOutOfRangeException exception if `a.Lenght < 4`...